src/psyclone/tests/psyir/nodes/routine_test.py

```diff
@@ -116,10 +116,10 @@ def test_routine_name_setter_preexisting_tag():
 
     # But it is fine if the name is the same
     node2 = Routine("hello", symbol_table=symtab)
```

Comment:
I think we need a check here that symtab has been deep-copied for the new Routine.

Reply:
Added checks here and in the constructor test